test/rubygems/test_gem_commands_setup_command.rb

```diff
@@ -74,6 +75,13 @@ def gem_install name
   end
 
   def test_execute_regenerate_binstubs
+    # Gem.ruby_version needs to be correct for passing test
+    if Gem.instance_variable_defined? :@ruby_version
```

Comment:
Why don't you use `util_set_RUBY_VERSION`?

And I'm not sure why this line was required. Can you describe this?

Reply:
> Why don't you use util_set_RUBY_VERSION?

I was unaware of it.  All the issue concerned was that one variable

> And I'm not sure why this line was required. Can you describe this?

There must have been another test that changed the value from default, and didn't reset it.  That's what I was speaking of in a previous message about trying to find those issues, which would require adding logging to `Gem::TestCase` and auditing all left-over artifacts...

Reply:
Investigated further.  See [test_gem_request_set_gem_dependency_api.rb:L35](https://github.com/rubygems/rubygems/blob/2ea667874491eadd9fcdcb6421795bfc52252ab1/test/rubygems/test_gem_request_set_gem_dependency_api.rb#L35)?

Also, reviewing that code, it appears that if `util_set_RUBY_VERSION` was called twice, it would remove the initial/real settings.  IOW, it seems to assume that every call to `util_set_RUBY_VERSION` is followed by a call to `util_restore_RUBY_VERSION`.  Maybe I'm missing something...